### ISSUE TITLE
The training is incorrectly setup, testing data is leaked during training, need to fix asap

### ISSUE THREAD

**Issue by RemyLau** [krishnanlab/node2vecplus_benchmarks#1]
https://github.com/krishnanlab/node2vecplus_benchmarks/blob/60449db21f6f71ce0e874d2aa57a8b4335188376/script/eval_gene_classification_gnn.py#L85-L87

Use only the training data for loss backpropagation

```python
out = model(data.x, data.adj)[train_idx]
loss = criterion(out, data.y[train_idx].to(torch.float))
```
